src/client/Microsoft.Identity.Client/AppConfig/PublicClientApplicationBuilder.cs

```diff
@@ -193,6 +195,27 @@ public PublicClientApplicationBuilder WithBroker(bool enableBroker = true)
 #pragma warning restore CS0162 // Unreachable code detected
         }
 
+        /// <summary>
+        /// Brokers enable Single-Sign-On, device identification,
+        /// and application identification verification. To enable one of these features,
+        /// you need to set the WithBroker() parameters to true. See https://aka.ms/msal-net-brokers 
+        /// for more information on platform specific settings required to enable the broker.
+        /// 
+        /// On iOS and Android, Authenticator and Company Portal serve as brokers.
+        /// On Windows, WAM (Windows Account Manager) serves as broker. See https://aka.ms/msal-net-wam
+        /// </summary>
+        /// <param name="brokerOptions">This provides cross platform options for broker.</param>
+        /// <returns>A <see cref="PublicClientApplicationBuilder"/> from which to set more
+        /// parameters, and to create a public client application instance</returns>
+        /// <remarks>If your app uses .NET classic or .NET Core 3.x, and you wish to use the Windows broker, 
+        /// please install the NuGet package Microsoft.Identity.Client.Desktop and call .WithDesktopFeatures()</remarks>
+        public PublicClientApplicationBuilder WithBroker(BrokerOptions brokerOptions)

```

Comment:
I'm not sure I understand the why this method is needed. If it's called on anything other than UWP and Net6_Win, it will throw.

If I want to specify options, is this the proper calling pattern then:
```csharp
builder.WithBroker(someOptions).WithWindowsBroker()
```
Or
```csharp
builder.WithWindowsBrokerOptions(options).WithWindowsBroker();
```

BrokerOptions also must have OS specified, this means two places where I have to specify OS.

I was under the impression that we would have either:
- a specific method for each OS - `WithWindowsBroker(WindowsBrokerOptions)` or
- a generic one - `WithBroker(brokerOptions)` - user specifies the OS here.

Reply:
We are going with  WithBroker(brokerOptions) - user specifies the OS here.